shepherd/Goal.py

```diff
@@ -1,41 +1,71 @@
-"""
-class Goal:
-
-    def __init__(self):
-        self.goal_state = 0 #unowned, blueowned, goldowned, bluebid, goldbid
-        self.bid_timer = timer thread(sends to shepherd main)
-        self.multiplier_timer
-        self.goal_multiplier
-        self.price
-        self.alliance = alliance_object
-
-class Alliance:
-
-    self.score
-    self.score_multiplier
-    self.increase_score()
-
-class Timer:
-    gameTime(gameState):
-        starts with StateChange(Active)
-        write to queue current time reamining
-        at end, writes statechange(idle)
-
-    sensorCooldown(side):
-        write to queue current time remaining
-        at end, sets both sensor multipiers to not used
-
+from Utils import *
+from Timer import *
 
+class Goal:
+    """This class holds the state values for the goal, such as its point value,
+       it's multipliers, and it's alliance.
+            name - A String holding the name of the goal
+            value - An Integer representing the inherent point value of the Goal
+            owner - An Alliance object representing the current owner of
+                    the goal
+            current_bid - An Integer representing the value of the current bid
+                          on this goal
+            current_bid_team - An Alliance object representing the team that
+                               has submited the most recent bid
+            previous_bid - An Integer representing the value of the previous
+                           bid
+            previous_bid_team - An Alliance object representing the team that
+                                made the previous bid
+            bid_timer - A timer tracking the time left to bid
+            gold_two_x_timer - A Timer tracking how long the 2x powerup applied
+                               by the gold team will last
+            gold_zero_x_timer - A Timer tracking how long the 0x powerup
+                                applied by the gold team will last
+            blue_two_x_timer - A Timer tracking how long the 2x powerup applied
+                               by the blue team will last
+            blue_zero_x_timer - A Timer tracking how long the 0x powerup applied
+                                by the blue team will last
+    """
+    bid_increase_constant = CONSTANTS.BID_INCREASE_CONSTANT
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+        self.owner = None
+        self.current_bid = 0
+        self.current_bid_team = None
+        self.previous_bid = 0
+        self.previous_bid_team = None
+        self.bid_timer = Timer('bid', name)
+        self.gold_two_x_timer = Timer('duration', name)
```

Comment:
You should make all of these (e.g. 'duration', 'cooldown', etc) also a class in Utils.py